keps/sig-architecture/20191015-conformance-pr-gate.md

```diff
@@ -0,0 +1,229 @@
+
+Check these off as they are completed for the Release Team to track. These checklist items _must_ be updated for the enhancement to be released.
+
+- [ ] kubernetes/enhancements issue in release milestone, which links to KEP (this should be a link to the KEP location in kubernetes/enhancements, not the initial KEP PR)
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+**Note:** Any PRs to move a KEP to `implementable` or significant changes once it is marked `implementable` should be approved by each of the KEP approvers. If any of those approvers is no longer appropriate than changes to that list should be approved by the remaining approvers and/or the owning SIG (or SIG-arch for cross cutting KEPs).
+
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://github.com/kubernetes/enhancements/issues
+[kubernetes/kubernetes]: https://github.com/kubernetes/kubernetes
+[kubernetes/website]: https://github.com/kubernetes/website
+
+## Summary
+
+
+We have a policy requiring Conformance Tests before promoting beta APIs to GA.
+This policy needs automation to ensure we eventually have coverage for
+everything. The proposal provides a method of measuring Conformance coverage and
+placing a `/hold` on any PR which fails to maintain or increase said coverage.
+
+
+## Motivation
+
+
+Currently PRs are merging that do not meet the Conformance Test policy
+requirement or accidentally decrease coverage by changes to how tests or
+frameworks are written.
+
+
+### Goals
+
+* Defining Conformance Coverage for API Operation 
+* Runnning Coverage per PR against kubernetes/kubernetes
+* Feedback with list of newly promoted API Operations lacking test hits
+* Feedback with list of previously tested API Operations lacking test hits
+* Soft gate with '/hold' (or similar merge blocking tag)
+* Hard gate 
+
+### Non-Goals
+
+* Defining Conformance Coverage for Watch Methods
+* Defining Conformance Coverage for Object Fields
+
+## Proposal
+
+The proposal consists of four deliverables:
+* prow-job that runs periodically/continously against master for baseline
+* prow-job that waits for PR e2e test jobs to finish for comparision
+* automated workflow for tagging / failing as a test for all PRs that reduce coverage percentage
+
```

Comment:
We need to understand how to address disjoint graduation to GA and graduation of e2e tests to conformance. See https://github.com/kubernetes/kubernetes/pull/81571#issuecomment-546514199